plugins/woocommerce/src/Admin/Features/ProductBlockEditor/Init.php

```diff
@@ -210,35 +218,130 @@ public function set_current_screen_to_block_editor_if_wc_admin() {
 
-		$editor_settings['templateEvents'] = array(
-			'product'           => $block_template_logger->get_formatted_template_events( 'simple-product' ),
-			'product_variation' => $block_template_logger->get_formatted_template_events( 'product-variation' ),
-		);
+		usort( $product_templates, function ( $a, $b ) {
+			return $a->get_order() - $b->get_order();
+		} );
+
+		$editor_settings['productTemplates'] = array_map( function ( $product_template ) {
+			return $product_template->to_JSON();
+		}, $product_templates );
 
 		$block_editor_context = new WP_Block_Editor_Context( array( 'name' => self::EDITOR_CONTEXT_NAME ) );
 
 		return get_block_editor_settings( $editor_settings, $block_editor_context );
 	}
 
+	/**
+	 * Get default product templates.
+	 * 
+	 * @param array @templates The initial templates.
+	 * @return array The default templates.
+	 */
+	public function get_product_templates( array $templates ) {
+		$templates[] = new ProductTemplate(
+			array(
+				'id'                 => 'standard-product-template',
+				'title'              => __( 'Standard product', 'woocommerce' ),
+				'description'        => __( 'A single physical or virtual product, e.g. a t-shirt or an eBook.', 'woocommerce' ),
+				'order'              => 10,
+				'icon'               => 'shipping',
+				'layout_template_id' => 'simple-product',
+				'product_data'       => array(
+					'type' => 'simple',
+				),
+			)
+		);
+		$templates[] = new ProductTemplate(
+			array(
+				'id'                 => 'grouped-product-template',
+				'title'              => __( 'Grouped product', 'woocommerce' ),
+				'description'        => __( 'A set of products that go well together, e.g. camera kit.', 'woocommerce' ),
+				'order'              => 20,
+				'icon'               => 'group',
+				'layout_template_id' => 'simple-product',
+				'product_data'       => array(
+					'type' => 'grouped',
+				),
+			)
+		);
+		$templates[] = new ProductTemplate(
+			array(
+				'id'                 => 'affiliate-product-template',
+				'title'              => __( 'Affiliate product', 'woocommerce' ),
+				'description'        => __( 'A link to a product sold on a different website, e.g. brand collab.', 'woocommerce' ),
+				'order'              => 30,
+				'icon'               => 'link',
+				'layout_template_id' => 'simple-product',
+				'product_data'       => array(
+					'type' => 'external',
+				),
+			)
+		);
+		$templates[] = new ProductTemplate(
+			array(
+				'id'                 => 'variable-product-template',
+				'title'              => __( 'Variable product', 'woocommerce' ),
+				'description'        => __( 'A product with variations like color or size.', 'woocommerce' ),
+				'order'              => 40,
+				'icon'               => null,
+				'layout_template_id' => 'simple-product',
+				'product_data'       => array(
+					'type' => 'variable',
+				),
+			)
+		);
+
+		$supported_product_types = array(
+			'simple',
+			'grouped',
+			'external',
+			'variable',
+		);
```

Comment:
Instead of hardcoding this, we should look at the templates and return a list of product types that are supported by at least one template.

And, we should do this check on the filtered result from `apply_filters( 'woocommerce_product_editor_product_templates' )` so that any custom product templates added are considered as well.

Reply:
Nope I think we should not do that. We defined the relation between the product and the product template only by using the template's id for a reason. Once you create the product base on the selected template to get the template given the product is not accurate. Same thing here when trying the get the template given only the product type. We never know which is the exact match for that.

Reply:
But, let's say an extension adds a new product template with a specific layout template.... the way we are creating "default" product templates, won't we also create one that falls back to the classic editor? I don't think that is what should be expected in this case. The extension explicitly set a layout template for their product template because they support the new product editor.

Reply:
> won't we also create one that falls back to the classic editor?

No, we should not and we "can not" since a new `class WC_Product_Demo extends WC_Product` (see point 9) needs to be created when registering a brand new product type, the user needs to perform 2 steps for that: 1 for the product type registration (like in legacy) and 2 for the product template registration and association. If the product template does refer to an existing product type the user only needs to do the template registration and association.

In case of doing what you suggested we need to infer not just the product type based on the product template but also do all the required steps (topic that can be covered in a different issue more carefully). Another problem is that inferring the product type is not going to be accurate, so the title shown in the dropdown type selector for the classic editor might be seen buggy when showing a non expected title for a particular product type.